libs/coin-modules/coin-cardano/src/api/getPools.ts

```diff
@@ -23,13 +23,25 @@ export async function fetchPoolList(
 export async function fetchPoolDetails(
   currency: CryptoCurrency,
   poolIds: Array<string>,
+  ledgerPoolIds?: Array<string>,
 ): Promise<APIGetPoolsDetail> {
+  const currentPoolIds = poolIds.length == 0 ? ledgerPoolIds : poolIds;
```

Comment:
What if `ledgerPoolIds` undefined? 🤔 
If `poolIds` length is 0, you might do twice the same request if the response is empty.

Reply:
Maybe this is a version that you want to use?
```js
export async function fetchPoolDetails(
  currency: CryptoCurrency,
  poolIds: Array<string>,
  ledgerPoolIds?: Array<string>,
): Promise<APIGetPoolsDetail> {
  const currentPoolIds = poolIds.length == 0 ? ledgerPoolIds : poolIds;
  const { data } = await network<APIGetPoolsDetail>({
    method: "GET",
    url: isTestnet(currency)
      ? `${CARDANO_TESTNET_API_ENDPOINT}/v1/pool/detail`
      : `${CARDANO_API_ENDPOINT}/v1/pool/detail`,
    params: { poolIds: currentPoolIds },
  });
  if (data.pools.length === 0 && ledgerPoolIds) {
    return fetchPoolDetails(currency, ledgerPoolIds);
  }
  return data;
}
```

Reply:
Yes actually that's what I wanted to do
Thx